OptimizelySDK/Utils/ExperimentUtils.cs

```diff
@@ -46,16 +47,34 @@ public static bool IsExperimentActive(Experiment experiment, ILogger logger)
         /// <returns>true if the user meets audience conditions to be in experiment, false otherwise.</returns>
         public static bool IsUserInExperiment(ProjectConfig config, Experiment experiment, UserAttributes userAttributes)
         {
-            var audienceIds = experiment.AudienceIds;
+            var audienceConditions = experiment.GetAudienceConditionsOrIds();
 
-            if (!audienceIds.Any())
+            // If there are no audiences, return true because that means ALL users are included in the experiment.
+            if (audienceConditions == null || !audienceConditions.Any())
                 return true;
 
             if (userAttributes == null)
                 userAttributes = new UserAttributes();
-            
-            var conditionEvaluator = new ConditionEvaluator();
-            return audienceIds.Any(id => conditionEvaluator.Evaluate(config.GetAudience(id).ConditionList, userAttributes).GetValueOrDefault());
+
+            var conditionTreeEvaluator = new ConditionTreeEvaluator();
+
+            bool? EvaluateConditionsWithUserAttributes(JToken condition)
```

Comment:
I didn't like inline functions should be written in this way. Please use lambda expression. 
var evaluateConditionsWithUserAttributes = (condition) = > { } should be like this.